### PR TITLE
puller: support to decrease ts for frontier

### LINKED ISSUE #667: Puller doesn't handle checkpoint ts regression of kv client correctly
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. What did you do? If possible, provide a recipe for reproducing the error.

Run CDC on TiKV cluster where there are many region splits and region merges

### 2. What did you expect to see?

CDC works well

### 3. What did you see instead?

CDC reports fatal error:

```
[FATAL] [puller.go:182] ["The CRTs must be greater than the resolvedTs"] 
```

This is caused by puller doesn't handle checkpoint_ts regression of kv client correctly. Suppose following procedure:
1. There are two adjacent region regionA `[startA, endA)` and regionB `[startB, endB)` where `endA == startB`, the resolved ts-es of them are 1 and 2 respectively. Also in puller, the resolved ts is 1 for `[startA, endA)` and 2 for `[startB, endB)`.
2. The two region merges to regionAB `[startA, endB)`, and the new checkpoint ts is the minimun one of the two old region, which is 1. Still in puller, the resolved ts is 1 for `[startA, endA)` and 2 for `[startB, endB)`.
3. The two region splits to regionA `[startA, endB)` and regionB `[startB, endB)`. Both them derives the checkpoint ts, 1, from regoinAB.
4. Then regionA advances its resolvedTs to 2. So regionA has resolvedTs = 2 and regionB has 1. However in puller, the range `[startA, endA)` is updated to 2 but it doesn't know `[startB, endB)` has regressed, so in puller both range has resovledTs = 2, therefore it pushes the overall resolvedTs = 2. However regionB, who was registered with checkpoint ts = 1, may be still sending events before ts 2, leading to the fatal error.
 
### 4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        master
        ```

    - TiCDC version (execute `cdc version`):

        ```
        master
        ```

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
fix #667 

### What is changed and how it works?
1. send resolved event when starting a single event feed
2. support to decrease ts for frontier

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by leoppro** [comment]
@amyangfei @bobotu PTAL

File: cdc/kv/client.go [hunk 1]
```diff
@@ -1207,19 +1218,20 @@ func (s *eventFeedSession) singleEventFeed(
 		case *cdcpb.Event_Error:
 			return checkpointTs, errors.Trace(&eventError{err: x.Error})
 		case *cdcpb.Event_ResolvedTs:
-			lastResolvedTs = x.ResolvedTs
 			if !initialized {
 				continue
 			}
+			if x.ResolvedTs <= checkpointTs {
```
[the remark this answers is not in the document]
Comment (leoppro) [reply cdc/kv/client.go:1224]: yes, tikv may sent a resolvedts which less than checkpointTs

**Comment by amyangfei** [comment]
Good job 👍

**Pull request merged**